slurm/train.slurm

```diff
@@ -8,58 +8,57 @@
 #SBATCH --err=./logs/%x-%j.err
 #SBATCH --requeue
 
-# Specific configuration optimized for the Hugging Face Compute Cluster
-# Be ye warned this may not work on other clusters!
 module load cuda/12.4
-
-
 set -x -e
 
 source ~/.bashrc
 source openr1/bin/activate
 echo "START TIME: $(date)"
+export GPUS_PER_NODE=8
 
 MODEL=$1
 TASK=$2
 CONFIG_SUFFIX=$3
 ACCELERATOR=$4
 OPTIONAL_ARGS=$5
+CONFIG_FILE=recipes/$MODEL/$TASK/config_$CONFIG_SUFFIX.yaml
+GRAD_ACC_STEPS=$(grep 'gradient_accumulation_steps' $CONFIG_FILE | awk '{print $2}')
+MODEL=$(grep 'model_name_or_path:' $CONFIG_FILE | awk '{print $2}')
+REVISION=$(grep 'model_revision:' $CONFIG_FILE |head -n 1 | awk '{print $2}')
 
-# Training setup
+# Distributed configuration
 NUM_NODES=$SLURM_NNODES
 GPUS_PER_NODE=8
 WORLD_SIZE=$(($NUM_NODES*$GPUS_PER_NODE))
-# Due to conflicts between Accelerate's DeepSpeed configs and Transformers' TrainingArguments, we need to parse the gradient accumulation steps from the config file to ensure they match
-CONFIG_FILE=recipes/$MODEL/$TASK/config_$CONFIG_SUFFIX.yaml
-GRAD_ACC_STEPS=$(grep 'gradient_accumulation_steps' $CONFIG_FILE | awk '{print $2}')
+NODELIST=($(scontrol show hostnames $SLURM_JOB_NODELIST))
+MASTER_ADDR=${NODELIST[0]}  # First node for main process
+MASTER_PORT=6000
+TRAIN_NODES=("${NODELIST[@]}")
 
-# Check if we are running vLLM during training to adjust the world size
-if grep -q 'use_vllm:\s*true' "$CONFIG_FILE"; then
+USE_VLLM="false"
+if [[ -f "$CONFIG_FILE" ]] && grep -qE '^\s*use_vllm:\s*true' "$CONFIG_FILE"; then
     USE_VLLM="true"
-else
-    USE_VLLM="false"
 fi
-
+# if using vllm
 if [[ "$USE_VLLM" == "true" ]]; then
-    WORLD_SIZE=$(($WORLD_SIZE - 1))
+     TRAIN_NODES=("${NODELIST[@]:0:$((NUM_NODES - 1))}")
+     VLLM_NODE=${NODELIST[-1]} # Last node
+     TP=$(python scripts/get_tensor_parallel_size.py --model_name $MODEL --revision $REVISION --default_tp $GPUS_PER_NODE)
+     WORLD_SIZE=$((WORLD_SIZE - GPUS_PER_NODE))
+     NUM_NODES=$((NUM_NODES - 1))
+     # Run `trl vllm-serve` on Group 2 (Node 4)
+     srun --nodes=1 --ntasks=1 --nodelist=$VLLM_NODE trl vllm-serve --model $MODEL --revision $REVISION --tensor_parallel_size $TP &
+
+     OPTIONAL_ARGS="$OPTIONAL_ARGS --vllm_server_host=$VLLM_NODE"
```

Comment:
Does this work? I thought `--vllm_server_host` had to be written in dotted decimal notation, but `VLLM_NODE` will be in dashed?

Reply:
Yes it works, I think the AWS DNS must resolve the URL to the ip address.